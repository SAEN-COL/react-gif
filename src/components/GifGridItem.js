import React from 'react'
import PropType from 'prop-types';
export const GifGridItem = ( { id, title, url } ) => {
  return (
    <div className="card animate__animated animate__fadeIn animate__delay-1s">
      <img src={ url } alt={ title } />
      <p> { title } </p>
    </div>
  )
};

GifGridItem.propTypes = {
  title: PropType.string.isRequired,
  url: PropType.string.isRequired,
}