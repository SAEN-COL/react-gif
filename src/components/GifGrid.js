
import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';


export const GifGrid = ( { category }) => {
  
  const { data:images, loading } = useFetchGifs( category );

  return (
    <>
      <h3 className="animate__animated animate__fadeIn"> { category } </h3>
      
      {loading && <p className="animate__animated animate__flash">Loadding</p> }
      <div className="card-grid">
          { images.map(
            image =>  
              <GifGridItem 
                key={ image.id } 
                {...image} />   // Los tres puntos es para indicar 
                                // que los key del objeto se pasan
                                // como parametros o props del componente
          )}
      </div>
    </>
  )
}
