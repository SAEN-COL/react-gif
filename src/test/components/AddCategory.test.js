import React from "react";
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';


describe('Pruebas en el componente <AddCategory />', () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={ setCategories } />);

  beforeEach( () => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);

  });

  test('Debe de mostrarse correctamente el componente', () => {
    expect( wrapper ).toMatchSnapshot();
  });
  
  test('Debería cambiar la caja de texto', () => {
    const input = wrapper.find('input');
    const value = 'Hola mundo';

    input.simulate('change', { target: {value: value} });
    expect( wrapper.find('p').text().trim() ).toBe( value );
  });

  test('No debe postear la información onSubmit', () => {
    wrapper.find('form').simulate('submit', { preventDefault(){} });
    
    expect( setCategories ).not.toHaveBeenCalled();
  });
  
  test('Debe llamar el setCategories y limpiar la caja de texto', () => {
    const value = 'Hola mundo';

    // 1. Simular el inputChange
    wrapper.find('input').simulate('change', { target: { value: value } });
    

    // 2. Simular el submit
    wrapper.find('form').simulate('submit', { preventDefault(){} });

    // 3. setCategories se debe haber llamado
    expect( setCategories ).toHaveBeenCalled(); // Que se haya llamado al menos una vez
    expect( setCategories ).toHaveBeenCalledTimes( 1 ); // La misma que el anterior, que se haya llamado al menos una sola vez
    expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) ); // Se espera que se llame con... en este caso una función.

    // 4. El valor del input debe estar ''
    expect( wrapper.find('input').prop('value') ).toBe('');
  });
  
});
