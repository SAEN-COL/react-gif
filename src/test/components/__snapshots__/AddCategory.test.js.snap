// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AddCategory /> Debe de mostrarse correctamente el componente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <p>
     
     
  </p>
</form>
`;
